crm_sale_lead_tag_sync/models/crm_lead.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2020 Iván Todorovich <ivan.todorovich@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class CrmLead(models.Model):
+    _inherit = 'crm.lead'
+
+    def _sync_tag_ids(self):
+        """ Synchronizes tag_ids to the sale orders. """
+        for rec in self:
+            for order in rec.order_ids:
+                if order.tag_ids != rec.tag_ids:
+                    order.tag_ids = rec.tag_ids
+
+    def write(self, vals):
+        res = super().write(vals)
+        if 'tag_ids' in vals:
```

Comment:
As I said, you can use inverse as it is more 'readable'.

`tag_ids = fields(inverse=_inverse_tag_ids

def _inverse_tag_ids(self):
    self._sync_tag_ids()
`

Reply:
I don't like to use inverse because it's decreases compatibility with other modules

Reply:
> I don't like to use inverse because it decreases compatibility with other modules

Why ?

Reply:
We're adding an inverse method to a field that already exist, and you can only have one inverse per field.
I know it's unlikely, but if some other module tries to do the same, only of them will work

It would be different if the field was created by our module, because in that case we'd own the field and any other module wanting to extend it can simply overload the inverse method.